engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
```

Comment:
I see your point Thomas but as chinmay pointed out shouldn't both this and checkpointed callback not be called for stateless. Gaurav if you use this call in stateless operator to persist your operator state in your own way there is no inverse callback from the platform to recover that state.